pkg/operator/events/recorder_upstream.go

```diff
@@ -27,20 +30,60 @@ type upstreamRecorder struct {
 	broadcaster       record.EventBroadcaster
 	eventRecorder     record.EventRecorder
 	involvedObjectRef *corev1.ObjectReference
+
+	// shuttingDown indicates that the broadcaster for this recorder is being shut down
+	shuttingDown  bool
+	shutdownMutex sync.RWMutex
+
+	// fallbackRecorder is used when the kube recorder is shutting down
+	// in that case we create the events directly.
+	fallbackRecorder Recorder
+}
+
+var eventsCounterMetric = metrics.NewCounterVec(&metrics.CounterOpts{
+	Subsystem:      "event_recorder",
+	Name:           "total_events_count",
+	Help:           "Total count of events processed by this event recorder per involved object",
+	StabilityLevel: metrics.ALPHA,
+}, []string{"namespace", "name"})
```

Comment:
@deads2k i think this will have reasonable cardinality and can show us how many events we receive per operator... i wonder if we want to break this into warnings vs. normal events :-)

i imagine an alert that can fire if we see "abnormal" amount of "warning" events in a period of time...

Reply:
cc @smarterclayton